src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -2924,27 +2924,140 @@ LogicalResult ONNXSplitV11Op::inferShapes(
   if (axisIndex < -inputRank || axisIndex >= inputRank)
     return onnx_mlir::Diagnostic::emitAttributeOutOfRangeError(
         *this->getOperation(), "axis", axisIndex,
         onnx_mlir::Diagnostic::Range<int64_t>(-inputRank, inputRank - 1));
+  if (axisIndex < 0)
+    axisIndex += inputRank;
+
+  Value splitValue = split();
+  if (splitValue.getType().isa<NoneType>()) {
+    // keepdims attribute is ignored unless split input is specified
+    int64_t keep = keepdims();
+    // keepdims must be 0 or 1
+    if (keep < 0 || keep > 1)
+      return onnx_mlir::Diagnostic::emitAttributeOutOfRangeError(
+          *this->getOperation(), "keepdims", keep,
+          onnx_mlir::Diagnostic::Range<int64_t>(0, 1));
+    return success();
+  }
+  auto splitType = splitValue.getType().cast<ShapedType>();
+  ArrayRef<int64_t> splitShape = splitType.getShape();
+  int64_t splitRank = splitShape.size();
+  if (splitRank > 1)
+    return emitOpError() << ": split has rank " << splitRank << " > 1";
+  if (DenseElementsAttr entries =
+          getDenseElementAttributeFromONNXValue(splitValue)) {
+    if (splitRank == 0) {
+      auto scalar = getScalarValue<int64_t>(entries, splitType);
+      if (scalar <= 0)
+        return emitOpError() << ": split scalar " << scalar << " <= 0";
+    } else {
+      int64_t sum = 0;
+      for (auto entry : entries.getValues<IntegerAttr>()) {
+        int64_t i = entry.getInt();
+        if (i < 0)
+          return emitOpError() << ": split tensor has entry " << i << " < 0";
+        sum += i;
+      }
+      int64_t dimSize = inputShape[axisIndex];
+      if (dimSize != -1 && dimSize != sum)
+        return emitOpError() << ": split tensor entries sum to " << sum
+                             << " != axis dimension size " << dimSize;
+    }
+  }
 
   return success();
 }
 
 LogicalResult ONNXSplitToSequenceOp::inferShapes(
     std::function<void(mlir::Region &)> doShapeInference) {
-  return emitError(NOT_IMPLEMENTED_MESSAGE);
+  Value inputValue = input();
+  if (!hasShapeAndRank(inputValue))
+    return success(); // Cannot infer output shape if input shape isn't known.
+
+  // NOTE: all the asserts below are conditions checked in verify()
+
+  auto inputType = inputValue.getType().cast<ShapedType>();
+  ArrayRef<int64_t> shape = inputType.getShape();
+  int64_t rank = shape.size();
+  int64_t axisIndex = axis();
+  assert((-rank <= axisIndex && axisIndex < rank) && "axis out of range");
+  if (axisIndex < 0)
+    axisIndex += rank;
+  int64_t dimSize = shape[axisIndex];
+
+  // start with length unknown and dims == shape with unknown dimension size
+  // for axis (-1 is ShapedType::kDynamicSize), and edit it as needed below
+  int64_t length = -1;
+  SmallVector<int64_t, 4> dims(shape.begin(), shape.end());
+  dims[axisIndex] = -1;
+
+  Value splitValue = split();
+  if (splitValue.getType().isa<NoneType>()) {
+    // keepdims attribute is ignored unless split input is specified
```

Comment:
Document says: If input 'split' is specified, this attribute is ignored.
It's easier for me to understand if the comment is: Since split is not specified, check the keepdims attribute

Reply:
done